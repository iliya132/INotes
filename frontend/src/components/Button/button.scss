@import '../../Misc/colors.scss';

.primary-button {
    background-color: $G3;
    border: none;
    border-radius: 37px;
    min-width: 170px;
    height: 41px;
    color: $Gray;
    font-size: 18px;
    font-weight: 500;
    line-height: 40px;
    text-align: center;
    font-style: normal;
    display: block;
    text-transform: lowercase;
    letter-spacing: 3px;
    padding-left: 15px;
    padding-right: 15px;
    cursor: pointer;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));
    user-select: none;
}
