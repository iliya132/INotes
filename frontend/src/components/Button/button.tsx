import classNames from 'classnames';
import React from 'react';
import styles from './button.scss';
import { IButtonProps } from './types';

export function Button(props: IButtonProps) {
    const { title, className, disabled, onClick, type } = props;
    return (
        <button className={classNames(styles['primary-button'], className)} disabled={disabled} onClick={onClick} type={type}>
            {title}
        </button>
    );
}
