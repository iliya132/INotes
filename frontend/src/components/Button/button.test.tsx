/* eslint-disable no-undef */
import React from 'react';
import { createRoot, Root } from 'react-dom/client';
import { Button } from './button';
import { act } from 'react-dom/test-utils';


let root: Root | null = null;
let container: Element | null = null;
beforeEach(() => {
    act(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        root = createRoot(container);
    });
});

afterEach(() => {
    act(() => {
        root!.unmount();
        container!.remove();
        container = null;
    });
});

describe('Test Button component', () => {
    it('can be rendered', () => {
        act(() => {
            root!.render(<Button title="test" />);
        });

        let btn = container?.querySelector('button');

        expect(btn!.textContent).toEqual('test');
        expect(btn).toMatchSnapshot('button-snapshot');
    });

    it('can call events on click', () => {
        let isPressed = false;
        const testCallback = () => (isPressed = true);
        act(() => {
            const button = <Button title="test" onClick={testCallback} />;
            button.props.onClick();
        });

        expect(isPressed).toEqual(true);
    });

    it('setup className', () => {
        const className = 'test-classname'
        const expectedClassName = 'primary-button test-classname';
        act(() => {
            root!.render(<Button title="test" className={className} />);
        });
        let btn = container?.querySelector('button');

        expect(btn!.className).toEqual(expectedClassName);
    });

    it('shows loading when isLoading', () => {
        act(() => {
            root!.render(<Button title="test" isLoading={true} />);
        });
        let btn = container?.querySelector('button');
        expect(btn!.childNodes[1]).toMatchSnapshot('button-loading-snapshot');
    });

    it('should be disabled when disabled props is set', () => {
        act(() => {
            root!.render(<Button title="test" disabled={true} />);
        });
        let btn = container?.querySelector('button');
        expect(btn!.outerHTML).toContain('disabled');
    });
});
