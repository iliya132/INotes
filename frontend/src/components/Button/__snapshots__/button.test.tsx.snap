// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component can be rendered: button-snapshot 1`] = `
<button
  class="primary-button"
>
  test
  <div
    class="button-content"
  />
</button>
`;

exports[`Test Button component shows loading when isLoading: button-loading-snapshot 1`] = `
<div
  class="button-content"
>
  <div
    style="fill: white; height: 64px; width: 64px;"
  >
    <svg
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      
  
      <path
        d="M0 12 V20 H4 V12z"
        transform="translate(0 0)"
      >
        
    
        <animatetransform
          attributeName="transform"
          begin="0"
          calcMode="spline"
          dur="1.5s"
          keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
          keyTimes="0;0.3;0.6;1"
          repeatCount="indefinite"
          type="translate"
          values="0 0; 28 0; 0 0; 0 0"
        />
        
  
      </path>
      
  
      <path
        d="M0 12 V20 H4 V12z"
        opacity="0.5"
        transform="translate(0 0)"
      >
        
    
        <animatetransform
          attributeName="transform"
          begin="0.1s"
          calcMode="spline"
          dur="1.5s"
          keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
          keyTimes="0;0.3;0.6;1"
          repeatCount="indefinite"
          type="translate"
          values="0 0; 28 0; 0 0; 0 0"
        />
        
  
      </path>
      
  
      <path
        d="M0 12 V20 H4 V12z"
        opacity="0.25"
        transform="translate(0 0)"
      >
        
    
        <animatetransform
          attributeName="transform"
          begin="0.2s"
          calcMode="spline"
          dur="1.5s"
          keySplines="0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8;0.2 0.2 0.4 0.8"
          keyTimes="0;0.3;0.6;1"
          repeatCount="indefinite"
          type="translate"
          values="0 0; 28 0; 0 0; 0 0"
        />
        
  
      </path>
      

    </svg>
    

  </div>
</div>
`;
